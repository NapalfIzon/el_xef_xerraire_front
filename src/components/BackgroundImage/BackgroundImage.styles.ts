import { StyleSheet } from "react-native";
import { colors } from "../../styles/root.styles";

export default StyleSheet.create({
  body: {
    flex: 1,
    width: "100%",
    height: "100%",
    position: "absolute",
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: colors.backgroundImage,
  },
  backgroundImage: {
    justifyContent: "center",
    height: 250,
    width: 250,
    marginBottom: 50,
    opacity: 0.2,
  },
});
