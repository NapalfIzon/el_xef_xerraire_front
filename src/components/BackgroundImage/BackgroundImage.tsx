import React from "react";
import { ImageBackground, View } from "react-native";
import styles from "./BackgroundImage.styles";

const BackgroundImage = () => {
  return (
    <View style={styles.body}>
      <ImageBackground
        source={{ uri: "https://i.ibb.co/qW91Q3W/background-logo.webp" }}
        resizeMode="cover"
        style={styles.backgroundImage}
      />
    </View>
  );
};

export default BackgroundImage;
