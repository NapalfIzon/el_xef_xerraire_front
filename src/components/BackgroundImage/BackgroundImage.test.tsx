import React from "react";
import { render } from "@testing-library/react-native";
import BackgroundImage from "./BackgroundImage";

describe("Given a BackgroundImage component,", () => {
  describe("When it is rendered,", () => {
    test("Then it should render a brackground image.", () => {
      const result = render(<BackgroundImage />);

      expect(result).toMatchSnapshot();
    });
  });
});
