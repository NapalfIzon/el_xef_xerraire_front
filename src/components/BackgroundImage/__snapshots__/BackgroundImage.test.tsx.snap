// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a BackgroundImage component, When it is rendered, Then it should render a brackground image. 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": 250,
        "justifyContent": "center",
        "marginBottom": 250,
        "width": 250,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "https://i.ibb.co/qW91Q3W/background-logo.webp",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 250,
            "width": 250,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
